crates/cargo-gpu/src/install.rs

```diff
@@ -7,16 +7,70 @@ use crate::spirv_source::{
 use crate::{cache_dir, spirv_source::SpirvSource};
 use anyhow::Context as _;
 use cargo_metadata::Metadata;
-use log::{info, trace};
 use spirv_builder::SpirvBuilder;
 use std::path::{Path, PathBuf};
 
+/// Represents a functional backend installation, whether it was cached or just installed.
+#[derive(Clone, Debug)]
+#[expect(
+    clippy::exhaustive_structs,
+    reason = "never adding private members to this struct"
+)]
+pub struct InstalledBackend {
+    /// path to the `rustc_codegen_spirv` dylib
+    pub rustc_codegen_spirv_location: PathBuf,
+    /// toolchain channel name
+    pub toolchain_channel: String,
+    /// directory with target-specs json files
+    pub target_spec_dir: PathBuf,
+}
+
+impl InstalledBackend {
+    /// Creates a new `SpirvBuilder` configured to use this installed backend.
+    #[expect(
+        clippy::missing_panics_doc,
+        clippy::expect_used,
+        reason = "unreachable"
```

Comment:
It would be good to have a very brief reason why it's unreachable.

Reply:
Changed it to this:

```rust
    /// Creates a new `SpirvBuilder` configured to use this installed backend.
    #[expect(
        clippy::unreachable,
        reason = "it's unreachable, no need to return a Result"
    )]
    #[expect(clippy::impl_trait_in_params, reason = "forwarding spirv-builder API")]
    #[inline]
    pub fn to_spirv_builder(
        &self,
        path_to_crate: impl AsRef<Path>,
        target: impl Into<String>,
    ) -> SpirvBuilder {
        let mut builder = SpirvBuilder::new(path_to_crate, target);
        let result = self.configure_spirv_builder(&mut builder);
        match result {
            Ok(()) => (),
            Err(_) => unreachable!("we set target before calling this function"),
        }
        builder
    }
```

Reply:
Extremely minor point but I think that can be reduced to: `self.configure_spirv_builder(&mut builder).unwrap_or_else(|_| unreachable...)`?

Reply:
Wait, are you mistaking my English question mark for the rust syntatic sugar, `?`?

Reply:
oops, applied your change